windows-installer/auto-update.ps1

```diff
@@ -0,0 +1,134 @@
+[CmdletBinding()]
+param()
+begin {}
+process {
+    New-EventLog -Source "certbot/auto-update.ps1" -LogName "CertbotAutoUpdate" -ErrorAction SilentlyContinue
+
+    function Write-Message($message, $level = "Information") {
+        Write-EventLog -Source "certbot/auto-update.ps1" -LogName "CertbotAutoUpdate" -EventID 1 -EntryType $level -Message $message
+        Write-Host $message
+    }
+
+    function Write-Error($message) {
+        Write-EventLog -Source "certbot/auto-update.ps1" -LogName "CertbotAutoUpdate" -EventID 1 -EntryType Error -Message $message
+        throw $message
+    }
+
+    Write-Message "Starting auto-update workflow ..."
+
+    $ErrorActionPreference = 'Stop'
+
+    $installDir = $PSScriptRoot
+
+    if ((Test-Path HKLM:\Software\Certbot) -And ((Get-ItemProperty -Path HKLM:\Software\Certbot).PSObject.Properties.Name -Contains "CertbotSigningPubKey")) {
+        $certbotSigningPubKey = (Get-ItemProperty -Path HKLM:\Software\Certbot).CertbotSigningPubKey
+    } else {
+        $certbotSigningPubKey = '
+-----BEGIN PUBLIC KEY-----
+MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6MR8W/galdxnpGqBsYbq
+OzQb2eyW15YFjDDEMI0ZOzt8f504obNs920lDnpPD2/KqgsfjOgw2K7xWDJIj/18
+xUvWPk3LDkrnokNiRkA3KOx3W6fHycKL+zID7zy+xZYBuh2fLyQtWV1VGQ45iNRp
+9+Zo7rH86cdfgkdnWTlNSHyTLW9NbXvyv/E12bppPcEvgCTAQXgnDVJ0/sqmeiij
+n9tTFh03aM+R2V/21h8aTraAS24qiPCz6gkmYGC8yr6mglcnNoYbsLNYZ69zF1XH
+cXPduCPdPdfLlzVlKK1/U7hkA28eG3BIAMh6uJYBRJTpiGgaGdPd7YekUB8S6cy+
+CQIDAQAB
+-----END PUBLIC KEY-----
+'
+    }
+
+    if ((Test-Path HKLM:\Software\Certbot) -And ((Get-ItemProperty -Path HKLM:\Software\Certbot).PSObject.Properties.Name -Contains "CertbotUpgradeApiURL")) {
+        $certbotUpgradeApiURL = (Get-ItemProperty -Path HKLM:\Software\Certbot).CertbotUpgradeApiURL
+    } else {
+        $certbotUpgradeApiURL = 'https://api.github.com/repos/certbot/certbot/releases/latest'
+    }
+
+    # Get current local certbot version
+    try {
+        $currentVersion = certbot --version
+        $currentVersion = $currentVersion -replace '^certbot (\d+\.\d+\.\d+).*$', '$1'
+        $currentVersion = [System.Version]"$currentVersion"
+    } catch {
+        Write-Message @"
+An error occured while fetching the current local certbot version:
+$_
+Assuming Certbot is not up-to-date.
+"@ "Warning"
+        $currentVersion = "0.0.0"
```

Comment:
Should this also be a `System.Version` object?

Reply:
Indeed!